vnext/Shared/Threading/BatchingQueueThread.cpp

```diff
@@ -62,9 +69,17 @@ void BatchingQueueThread::runOnQueueSync(std::function<void()> && /*func*/) noex
 }
 
 void BatchingQueueThread::quitSynchronous() noexcept {
-  // Used by OInstance
-  // assert(false && "Not supported");
-  // std::terminate();
+  Mso::ManualResetEvent resetEvent;
+  {
+    std::lock_guard<std::mutex> lck(m_mutex);
+
+    m_quit = true;
+
+    if (m_taskQueue)
+      m_taskQueue->emplace_back([&resetEvent]() { resetEvent.Set(); });
+  }
+
+  resetEvent.Wait();
```

Comment:
>resetEvent.Wait(); [](start = 2, length = 18)

Ideally we should wait only if the current queue does not have the thread access. Otherwise, it is going to be a deadlock. Can we do the m_queueThread.quitSynchronous() instead? We should call the onBatchComplete() before that.